File: apps/problems/remove-linked-list-elements/boilerplate-full/function.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>
#include <climits>


struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(nullptr) {}
};

//##USER_CODE_HERE


int main() {
    
int n;
std::cin >> n;
ListNode* head = nullptr;
ListNode* tail = nullptr;
for (int i = 0; i < n; ++i) {
    int val;
    std::cin >> val;
    ListNode* newNode = new ListNode(val);
    if (head == nullptr) {
        head = newNode;
        tail = newNode;
    } else {
        tail->next = newNode;
        tail = newNode;
    }
}

int val;
std::cin >> val;
    Solution obj;
    ListNode* result = obj.removeElements(head, val);
    
ListNode* current = result;
while (current) {
  std::cout << current->val;
  if (current->next) std::cout << " ";
  current = current->next;
}
std::cout << std::endl;
    return 0;
}

File: apps/problems/remove-linked-list-elements/boilerplate-full/function.java

    import java.io.*;
    import java.util.*;

    
class ListNode {
    int val;
    ListNode next;
    ListNode(int x) { val = x; }
}


    //##USER_CODE_HERE##

    public class Main {
      
    public static void main(String[] args) throws IOException {
      BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
      
      int n_head = Integer.parseInt(br.readLine().trim());
      String[] listData_head = br.readLine().trim().split("\\s+");
      ListNode head = buildListNode(listData_head);
      
int val = Integer.parseInt(br.readLine().trim());
      Solution obj = new Solution();
      ListNode result = obj.removeElements(head, val);
      printListNode(result);
      br.close();
    }
      
  static ListNode buildListNode(String[] data) {
    ListNode dummy = new ListNode(0);
    ListNode current = dummy;
    for (String s : data) {
      current.next = new ListNode(Integer.parseInt(s));
      current = current.next;
    }
    return dummy.next;
  }

  static void printListNode(ListNode head) {
    ListNode current = head;
    while (current != null) {
      System.out.print(current.val);
      if (current.next != null) {
        System.out.print(" ");
      }
      current = current.next;
    }
    System.out.println();
  }

    }
  